lib/tzinfo/time_or_datetime.rb

```diff
@@ -255,6 +255,29 @@ def -(seconds)
       self + (-seconds)
     end
 
+    # Sets offset from UTC for the TimeOrDateTime. Returns a new TimeOrDateTime,
+    # preserving original class. Does NOT changes values of hour-min-second,
+    # so, (18:00 UTC).offset(3600) will became (18:00 +01:00).
+    # Considers original value's UTC offset wisely.
+    def offset(seconds)
```

Comment:
This seems to produce a different result depending on whether `DateTime` or `Time` is used to initialize the `TimeOrDateTime`:

```ruby
2.3.1 :023 > TZInfo::TimeOrDateTime.new(Time.new(2016,5,30,18,0,0,'+01:00'), false).offset(3600)
 => #<TZInfo::TimeOrDateTime: 2016-05-30 19:00:00 +0100> 
2.3.1 :024 > TZInfo::TimeOrDateTime.new(DateTime.new(2016,5,30,18,0,0,'+01:00'), false).offset(3600)
 => #<TZInfo::TimeOrDateTime: #<DateTime: 2016-05-30T18:00:00+01:00 ((2457539j,61200s,0n),+3600s,2299161j)>>
```